src/components/Drawer/Drawer.tsx

```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+
+const DrawerBackground = ({ isOpen, onClose }: { isOpen: boolean; onClose: () => void }) => (
+  <div className={`Layer__drawer-background ${isOpen ? 'open' : ''}`} onClick={onClose}></div>
+);
+
+export const Drawer = ({ isOpen, onClose, children }: { isOpen: boolean; onClose: () => void; children: React.ReactNode }) => {
+  return (
```

Comment:
You can use `classnames` here

`const drawerClassName = classNames(
      'Layer__drawer',
     isOpen && 'Layer__drawer--is-open'
    )`

And put it to div element `div className={drawerClassName}`